### PR TITLE
workflows/release: use PyPI trusted publisher

### LINKED ISSUE #86: Migrate to PyPI Trusted publishers
The Qiskit's PyPI account is enabling 2FA on Oct 19. As a consequence, deploying this repo won longer be able to use password authentication for publishing this package to PyPI after this date. The best mechanism PyPI offers for publishing now is trusted publishers:

https://docs.pypi.org/trusted-publishers/using-a-publisher/

which authorizes a GitHub Action workflow to push to PyPI and bypasses the need for any user credentials. 
So, this repo needs such a workflow. Take a look to other project as examples:
 * `.github/workflows/wheels.yml` in https://github.com/Qiskit/qiskit/pull/10999/files
 * `.github/workflows/deploy-code.yml` in https://github.com/qiskit-community/qiskit-machine-learning/pull/698/files

Once done, [a manual linking process needs to be done in the web UI](https://docs.pypi.org/trusted-publishers/adding-a-publisher/) by @mtreinish  .

### PR DESCRIPTION
**Pull request by airwoodix**
### Summary

Resolves #86.

@mtreinish could you please add this repository as a trusted publisher for [qiskit-aqt-provider](https://pypi.org/project/qiskit-aqt-provider/)? or, better, add the [aqt-admin](https://pypi.org/user/aqt-admin/) as co-maintainer of the project, such that we can administrate the PyPI project directly? Thanks!

### CODE REVIEW

**Review by fg-aqt — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]
This LGTM, I'm planning to hopefully migrate the pypi account on Wednesday (tomorrow), so as long as you aren't planning any releases before then we can merge this whenever (it only will work after the migration though).

**Comment by mtreinish** [comment]
Just a heads up I took care of the migration earlier today, so you should be good to use the trusted publisher mechanism now. I also sent an invite to the aqt-admin user to be a co-owner on the project.

**Comment by airwoodix** [comment]
Awesome, thanks a lot! We'll test this with the next release.

**Pull request merged**
